test/server/cards/03_TWI/units/CloneDiveTrooper.spec.ts

```diff
@@ -0,0 +1,33 @@
+describe('Clone Dive Trooper', function() {
+    integration(function(contextRef) {
+        it('Clone Dive Trooper\'s constant Coordinate ability should give -2/0 to target when attacking a unit', function () {
+            contextRef.setupTest({
+                phase: 'action',
+                player1: {
+                    groundArena: ['clone-dive-trooper', 'battlefield-marine', 'compassionate-senator'],
+                },
+                player2: {
+                    groundArena: ['pyke-sentinel', 'greedo#slow-on-the-draw']
+                }
+            });
+
+            const { context } = contextRef;
+
+            // Coordinate online
+            context.player1.clickCard(context.cloneDiveTrooper);
+            context.player1.clickCard(context.pykeSentinel);
+            expect(context.cloneDiveTrooper).toBeInZone('groundArena');
+            expect(context.cloneDiveTrooper.damage).toBe(0);
+            expect(context.pykeSentinel.damage).toBe(2);
+            context.player2.passAction();
+
+            context.player1.moveCard('battlefield-marine', 'discard');
```

Comment:
Try to use `moveCard` as little as possible, it is always better to use a game effect if possible. The reason is that `moveCard` doesn't necessarily trigger game events correctly, so it could cause unexpected game state that could mess with the test results. The most common use of it is for returning events from discard to hand so they could be used multiple times in one test.

In this case, better to defeat the marine with an attack or an event card like vanquish.